### PR TITLE
add/switch chain e2e tests

### PR DESCRIPTION
**Pull request by PeterYinusa**
## Explanation

Adds test coverage for `wallet_addEthereumChain` and `wallet_switchEthereumChain`, 

## More information

See https://github.com/MetaMask/metamask-extension/pull/14272

## Manual testing steps

yarn setup
yarn build:test
yarn test:e2e:single test/e2e/tests/chain-interactions.spec.js --browser=chrome
yarn test:e2e:single test/e2e/tests/chain-interactions.spec.js --browser=firefox

### CODE REVIEW

**Comment by seaona** [comment]
Looks good! Maybe we could also check these flows:
- Click Add Network + Cancel --> check that Network has not been added
- Click Add Network (with the Network already added and active) --> check nothing happens
- Click Add Network (with the Network already added and not active) --> check Switch network prop appears

They could also be included in another PR @PeterYinusa

**Review by seaona — DISMISSED** [review]

File: test/e2e/tests/chain-interactions.spec.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        await driver.clickElement({ text: 'Approve', tag: 'button' });
+        await driver.clickElement({ text: 'Switch network', tag: 'button' });
+
+        // switch to extension
+        await driver.waitUntilXWindowHandles(2);
+        await driver.switchToWindow(extension);
+
+        // verify current network
+        const networkDisplay = await driver.findElement('.network-display');
+        assert.equal(await networkDisplay.getText(), 'xDAI Chain');
```
Comment (digiwand) [inline test/e2e/tests/chain-interactions.spec.js:110]: I believe we can remove await here 
```suggestion
        assert.equal(networkDisplay.getText(), 'xDAI Chain');
```

File: test/e2e/tests/chain-interactions.spec.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        await driver.clickElement({ text: 'Cancel', tag: 'button' });
+
+        // switch to extension
+        await driver.waitUntilXWindowHandles(2);
+        await driver.switchToWindow(extension);
+
+        // verify networks
+        const networkDisplay = await driver.findElement('.network-display');
+        await networkDisplay.click();
+        assert.equal(await networkDisplay.getText(), 'Localhost 8545');
```
Comment (digiwand) [inline test/e2e/tests/chain-interactions.spec.js:57]: should be able to remove `await` here along with the other `await... getText()` instances

```suggestion
        assert.equal(networkDisplay.getText(), 'Localhost 8545');
```

File: test/e2e/tests/chain-interactions.spec.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
+const { strict: assert } = require('assert');
+const { convertToHexValue, withFixtures } = require('../helpers');
+
+describe('Chain Interactions', function () {
+  it('should add the XDAI chain', async function () {
```
Comment (digiwand) [inline test/e2e/tests/chain-interactions.spec.js:5]:
```suggestion
  it('should add the XDAI chain and not switch the network', async function () {
```

File: test/e2e/tests/chain-interactions.spec.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        const xDaiChain = await driver.findElements({
+          text: 'xDAI Chain',
+          tag: 'span',
+        });
+        assert.ok(xDaiChain.length, 1);
+      },
+    );
+  });
+
+  it('should switch to the XDAI chain', async function () {
```
Comment (digiwand) [inline test/e2e/tests/chain-interactions.spec.js:67]:
```suggestion
  it('should add the XDAI chain and switch the network', async function () {
```

**Review by digiwand — COMMENTED** [review]
Looks good to me! Just added some optional suggestions

File: test/e2e/tests/chain-interactions.spec.js (hunk 3 above)
Comment (PeterYinusa) [reply test/e2e/tests/chain-interactions.spec.js]: done

**Review by seaona — APPROVED** [review]

File: test/e2e/tests/chain-interactions.spec.js (hunk 1 above)
Comment (digiwand) [reply test/e2e/tests/chain-interactions.spec.js:110]: ah I didn't realize it was a promise. Thanks for clarifying this!

**Review by digiwand — APPROVED** [review]
TIL `getText()` from the `selenium-webdriver`  returns a promise. Thanks @PeterYinusa! LGTM! 

ref: https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/index_exports_WebElementPromise.html

**Pull request merged**